R/summary-log.R

```diff
@@ -191,14 +196,37 @@ extract_heuristics <- function(.s) {
   return(.out)
 }
 
-#' @describeIn extract_from_summary Extract objective function value
+#' @describeIn extract_from_summary Extract objective function value (without constant) for the final estimation method
 #' @importFrom purrr map map_dbl
-#' @param .subfield The field to extract from within "ofv". Defaults to the objective function value with no constant added, but the other fields are available too.
-extract_ofv <- function(.s, .subfield = c("ofv_no_constant", "ofv_with_constant", "ofv")) {
-  .subfield <- match.arg(.subfield)
-  .ofv <- map(.s, "ofv")
-  .out <- map_dbl(.ofv, .subfield, .default = NA_real_)
+extract_ofv <- function(.s) {
```

Comment:
1. maybe... I don't feel the need to export them unless someone asks for them. They were written as helpers for `summary_log_impl()`. A user can get any of this manually from the `bbi_nonmem_summary` object. I could see the argument that these helpers could be nice for a user too, so that they don't have to explore that object... but as you say, that would probably mean we write tests for them. I don't think it's worth it at this point.

2. Probably a good move. I know you love assert, but inherits is vectorized and I wrote some helpers to do this sort of thing for models like so https://github.com/metrumresearchgroup/rbabylon/blob/master/R/utils.R#L322 . I can add another for summaries too, if you can stomach that.

Also, I checked and `assert_class` won't work for multiple options like `inherits` does:

> classes | [character]Class names to check for inheritance with inherits. `x` must inherit from _all_ specified classes. _(emphasis added)_

Reply:
1. I phrased my original point (1) poorly. I agree that they should not be exported (at least for now). Should they also be marked as `@internal`?

2. I was trying to say that we could have an abstract class for summary objects corresponding to different model types, and then the assertion could be that `.s` inherits from that abstract class. I agree that the behavior of `assert_class()` is as you describe. But, I'm fine kicking the can on this for now.

Reply:
1. yes, good point. I'll mark them `@internal`
2. I see your point, but let's kick that can for now. I will add a helper to check, for consistency's sake. That will make the refactor easier later if/when we decide to.

Reply:
Ok so I started to implement part 2 and realized they don't actually act on a summary, they act on a _list_ of summaries (specifically a [column in a tibble](https://github.com/metrumresearchgroup/rbabylon/blob/develop/R/summary-log.R#L113)) and then I realized "oh wait, sometimes one or more of the things in that list are _not_ summary objects, if the summary call error'd..."

In short: 

* I'm not going to implement the check
* we _definitely_ shouldn't export these because they are very specific helpers for this exact case (which is what they were written as originally)